pkg/catalogsourceconfig/update.go

```diff
@@ -62,10 +61,17 @@ func (r *updateReconciler) Reconcile(ctx context.Context, in *v1alpha1.CatalogSo
 // deleteObjects deletes the CatalogSource in the old TargetNamespace.
 func (r *updateReconciler) deleteObjects(in *v1alpha1.CatalogSourceConfig) error {
 	cachedCSCSpec, found := r.cache.Get(in)
-	// This should never happen as it is because the cached Spec has changed
+	// This generally won't happen as it is because the cached Spec has changed
 	// that we are in the update reconciler.
+	// However, in the case that the cache is invalidated because of a restart,
+	// we could simply be unsure if the targetNamespace changed. Ideally,  we
+	// we should be using something like a direct reference to the old object
+	// instead of an in memory cache, since when the service restarts we will be
+	// unable to find the old object if the targetNamespace changed while we were
+	// down. For now, we will just ignore it.
```

Comment:
Like we discussed offline, this doesn't seem to be the issue @emmajiafan is talking about in the BZ.

Reply:
This is actually not the case, we are seeing this error because the cache error forces the csc into a bad state that won't be resolved unless the status is dropped. By preventing this bad state, we won't have to wait for the syncer to help us.